src/path/mod.rs

```diff
@@ -750,6 +750,22 @@ pub trait IntoDatabasePath<'a> {
 }
 
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct DatabaseViewPathRef<'a> {
+    db_name: DatabaseNameRef<'a>,
+    view_name: ViewNameRef<'a>,
+}
+
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct DatabaseViewPath {
+    db_name: DatabaseName,
+    view_name: ViewName,
+}
```

Comment:
The `DatabaseViewPath` (and `DatabaseViewPathRef`) types represent a URI path of the form `/db/_all_docs`. If I understand this correctly, then `view_name` should _always_ be the constant `_all_docs`, and hence `view_name` is redundant because it doesn't store any information. I suggest removing the `view_name` field and modifying iteration to always return `_all_docs` for the <q>view name</q> path segment.